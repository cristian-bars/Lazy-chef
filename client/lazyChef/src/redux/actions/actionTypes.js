export default {
  LOAD_RECIPES: 'LOAD_RECIPES',
  DELETE_RECIPE: 'DELETE_RECIPE',
  ADD_RECIPE: 'ADD_RECIPE',
  UPDATE_RECIPE: 'UPDATE_RECIPE',
  LOAD_RECIPE: 'LOAD_RECIPE',

  LOAD_USERS: 'LOAD_USERS',
  DELETE_USER: 'DELETE_USER',
  ADD_USER: 'ADD_USER',
  UPDATE_USER: 'UPDATE_USER',
  LOAD_USER: 'LOAD_USER',
};
