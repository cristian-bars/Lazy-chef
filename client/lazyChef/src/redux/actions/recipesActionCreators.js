import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/recipes';
const url = 'http://192.168.0.29:2022/recipes';
//const url = 'http://192.168.1.130:2022/recipes';

export function loadRecipes() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_RECIPES,
        recipes: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPES_ERROR',
      });
    }
  };
}

export function addRecipe(recipe) {
  return {
    type: actionTypes.ADD_RECIPE,
    recipe,
  };
}

export function deleteRecipe(recipeId) {
  return async dispatch => {
    await axios.delete(`${url}/${recipeId.id}`);
    dispatch({
      type: actionTypes.DELETE_RECIPE,
      recipeId,
    });
  };
}

export function updateRecipe(recipe) {
  return {
    type: actionTypes.UPDATE_RECIPE,
    recipe,
  };
}

export function loadRecipe(recipe) {
  return {
    type: actionTypes.LOAD_RECIPE,
    recipe,
  };
}

export function getRecipeById(recipeId) {
  console.log(`${url}/${recipeId}`);
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/${recipeId}`);
      dispatch({
        type: actionTypes.LOAD_RECIPE,
        recipe: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPE_ERROR',
      });
    }
  };
}
