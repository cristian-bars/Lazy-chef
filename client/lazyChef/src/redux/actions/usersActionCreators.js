import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/users';
const url = 'http://192.168.0.29:2022';
//const url = process.env.REACT_APP_PORT;

export function loadUsers() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_USERS,
        users: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}

export function addUsers(user) {
  return {
    type: actionTypes.ADD_USER,
    user,
  };
}

export function deleteUser(userId) {
  return async dispatch => {
    await axios.delete(`${url}/${userId.id}`);
    dispatch({
      type: actionTypes.DELETE_USER,
      userId,
    });
  };
}

export function updateUser(user) {
  return {
    type: actionTypes.UPDATE_USER,
    user,
  };
}

export function loadUser(user) {
  return {
    type: actionTypes.LOAD_USER,
    user,
  };
}

export function getUserById(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/login`, user);
      dispatch({
        type: actionTypes.LOAD_USER,
        userToken: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}
