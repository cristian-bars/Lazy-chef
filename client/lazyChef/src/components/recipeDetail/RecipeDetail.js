/* eslint-disable react-hooks/exhaustive-deps */
import React, {useEffect} from 'react';
import {View, Text, Image} from 'react-native';
import {connect} from 'react-redux';
import {PropTypes} from 'prop-types';
import {getRecipeById} from '../../redux/actions/recipesActionCreators';
import styles from './recipeDetailStyles';

const RecipeDetail = ({recipe, dispatch, route}) => {
  const {recipeId} = route.params;
  useEffect(() => {
    dispatch(getRecipeById(recipeId));
  }, [recipeId]);
  console.log(recipe);

  return (
    <View style={styles.container}>
      {recipe ? (
        <>
          <View>
            {recipe.image[0] ? (
              <Image
                style={styles.recipeImage}
                source={{uri: recipe.image[0]}}
              />
            ) : (
              <Image
                style={styles.recipeImage}
                source={{
                  uri: 'https://i.ibb.co/PwFdwdH/Vector-artistic-pen-and-ink-drawing-illustration-of-empty-plate-knife-and-fork.jpg',
                }}
              />
            )}
          </View>

          <Text style={styles.titleText}>{recipe.title}</Text>
          <Text style={styles.descriptionText}>{recipe.Description}</Text>
          <View style={styles.recipeIcons}>
            <View style={styles.recipeInfo}>
              <Image
                style={styles.imageIcons}
                source={{
                  uri: 'https://i.ibb.co/PwFdwdH/Vector-artistic-pen-and-ink-drawing-illustration-of-empty-plate-knife-and-fork.jpg',
                }}
              />
              <Text style={styles.iconsDetail}>
                {recipe.recipeIngredient.length}
              </Text>
            </View>
            <View style={styles.recipeInfo}>
              <Image
                style={styles.imageIcons}
                source={{
                  uri: 'https://i.ibb.co/PwFdwdH/Vector-artistic-pen-and-ink-drawing-illustration-of-empty-plate-knife-and-fork.jpg',
                }}
              />
              <Text style={styles.iconsDetail}>{recipe.difficulty}</Text>
            </View>
            <View style={styles.recipeInfo}>
              <Image
                style={styles.imageIcons}
                source={{
                  uri: 'https://i.ibb.co/PwFdwdH/Vector-artistic-pen-and-ink-drawing-illustration-of-empty-plate-knife-and-fork.jpg',
                }}
              />
              <Text style={styles.iconsDetail}>{recipe.totalTime}</Text>
            </View>
          </View>
        </>
      ) : (
        <Text>No hi ha recepta a carregar</Text>
      )}
    </View>
  );
};

RecipeDetail.propTypes = {
  recipe: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  dispatch: PropTypes.func.isRequired,
};

function mapStateToProps(store) {
  return {
    recipe: store.recipe,
  };
}

export default connect(mapStateToProps)(RecipeDetail);
