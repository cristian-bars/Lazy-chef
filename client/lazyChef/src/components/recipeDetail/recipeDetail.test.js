import 'react-native';
import React from 'react';
import RecipeDetail from './index';
import thunk from 'redux-thunk';
import Store from 'redux-mock-store';
import * as actions from '../../redux/actions/usersActionCreators';
import {Provider} from 'react-redux';
import {render, fireEvent} from '@testing-library/react-native';

describe('Given a RecipeDetail component', () => {
  const mockStore = Store([thunk]);
  let navigation;
  let myStore;
  let route;
  let ingredient;
  beforeEach(() => {
    jest.spyOn(actions, 'updateUser').mockReturnValueOnce({type: ''});
    myStore = mockStore({
      recipe: {
        _id: '1234abcd',
        title: 'macarrones',
        description: '1a',
        image: ['a'],
        recipeIngredient: ['b'],
      },
      userAcces: {
        user: {
          favouriteRecipes: ['1234abcd'],
        },
      },
    });
    navigation = {
      navigate: jest.fn(),
      goBack: jest.fn(),
    };
    route = {
      params: {
        recipeId: '1234',
      },
    };
  });

  test('renders correctly', () => {
    console.log(route.params);
    const wrapper = render(
      <Provider store={myStore}>
        <RecipeDetail route={route} navigation={navigation} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('When star button is pressed', () => {
    test('Then removeFav is invoked', () => {
      myStore = mockStore({
        userAcces: {
          user: {
            favouriteRecipes: ['1234abcd'],
          },
        },
        recipe: {
          _id: '1234abcd',
          title: 'macarrones',
          description: '1a',
          image: ['a'],
          recipeIngredient: ['b'],
        },
      });
      const {getByTestId} = render(
        <Provider store={myStore}>
          <RecipeDetail route={route} navigation={navigation} />
        </Provider>,
      );
      const favorite = getByTestId('favoritePress');
      fireEvent.press(favorite);

      expect(actions.updateUser).toHaveBeenCalled();
    });

    test('Then addFav is invoked', () => {
      myStore = mockStore({
        userAcces: {
          user: {
            favouriteRecipes: ['1234'],
          },
        },
        recipe: {
          _id: '1234abcd',
          title: 'macarrones',
          description: '1a',
          image: ['a'],
          recipeIngredient: ['b'],
        },
      });
      const {getByTestId} = render(
        <Provider store={myStore}>
          <RecipeDetail route={route} navigation={navigation} />
        </Provider>,
      );
      const favorite = getByTestId('favoritePress');
      fireEvent.press(favorite);

      expect(actions.updateUser).toHaveBeenCalled();
    });
  });
});
