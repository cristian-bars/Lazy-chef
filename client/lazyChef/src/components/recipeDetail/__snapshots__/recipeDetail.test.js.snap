// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a RecipeDetail component renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#D3E0FF",
    }
  }
>
  <View>
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 25,
            "height": 10,
            "left": 20,
            "opacity": 1,
            "padding": 20,
            "position": "absolute",
            "top": 20,
            "width": 10,
            "zIndex": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/img/back.png",
            }
          }
          style={
            Object {
              "height": 20,
              "left": -12,
              "position": "relative",
              "top": -10,
              "width": 20,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 25,
            "height": 10,
            "left": 330,
            "opacity": 1,
            "padding": 20,
            "position": "absolute",
            "top": 20,
            "width": 10,
            "zIndex": 1,
          }
        }
        testID="favoritePress"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/img/estrella_groga.png",
            }
          }
          style={
            Object {
              "height": 20,
              "left": -10,
              "position": "relative",
              "top": -10,
              "width": 20,
            }
          }
        />
      </View>
      <Image
        source={
          Object {
            "uri": "a",
          }
        }
        style={
          Object {
            "height": 300,
            "position": "relative",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "bold",
            "paddingBottom": 10,
          }
        }
      >
        macarrones
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
            },
            Object {
              "backgroundColor": "white",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          >
            Ingredientes
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/img/Ingredientes.jpg",
              }
            }
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          >
            1
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          >
            Dificultad
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/img/herramientas-y-utensilios.png",
              }
            }
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          >
            Tiempo total
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/img/duration.png",
              }
            }
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
            },
            Object {
              "backgroundColor": "lightgrey",
              "marginTop": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
              "width": 120,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#616161",
                "fontSize": 20,
                "fontWeight": "bold",
                "paddingVertical": 5,
              }
            }
          >
            Ingredientes
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
              "width": 120,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FF914D",
                "fontSize": 20,
                "fontWeight": "bold",
                "paddingVertical": 5,
              }
            }
          >
            Pasos
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <RCTScrollView
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
